### PR TITLE
bump: Torch `2.5.0`

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

aligning with the recent release of PT 2.5

<details>
  <summary>Before submitting</summary>

- [x] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--2790.org.readthedocs.build/en/2790/

### CODE REVIEW

File: .github/workflows/ci-tests.yml [hunk 1]
```diff
@@ -42,7 +42,7 @@ jobs:
           - "2.1.2"
           - "2.2.2"
           - "2.3.1"
-          - "2.4.0"
+          - "2.5.0"
```
Comment (Borda) [inline .github/workflows/ci-tests.yml:45]:
```suggestion
          - "2.4.1"
          - "2.5.0"
```

**Comment by Borda** [comment]
```
FAILED unittests/bases/test_collections.py::test_metric_collection_prefix_postfix_args[None-None] - AssertionError: assert False
 +  where False = isinstance(dict_keys(['DummyMetricSum', 'DummyMetricDiff']), <class 'odict_keys'>)
 +    where dict_keys(['DummyMetricSum', 'DummyMetricDiff']) = keys(keep_base=True)
 +      where keys = MetricCollection(\n  (DummyMetricSum): DummyMetricSum()\n  (DummyMetricDiff): DummyMetricDiff(),\n  prefix=new_prefix_\n).keys
 +    and   <class 'odict_keys'> = type(odict_keys(['new_prefix_DummyMetricSum', 'new_prefix_DummyMetricDiff']))
 +      where odict_keys(['new_prefix_DummyMetricSum', 'new_prefix_DummyMetricDiff']) = keys(keep_base=False)
 +        where keys = MetricCollection(\n  (DummyMetricSum): DummyMetricSum()\n  (DummyMetricDiff): DummyMetricDiff(),\n  prefix=new_prefix_\n).keys
FAILED unittests/bases/test_collections.py::test_metric_collection_prefix_postfix_args[prefix_-None] - AssertionError: assert False
 +  where False = isinstance(dict_keys(['DummyMetricSum', 'DummyMetricDiff']), <class 'odict_keys'>)
 +    where dict_keys(['DummyMetricSum', 'DummyMetricDiff']) = keys(keep_base=True)
 +      where keys = MetricCollection(\n  (DummyMetricSum): DummyMetricSum()\n  (DummyMetricDiff): DummyMetricDiff(),\n  prefix=new_prefix_\n).keys
 +    and   <class 'odict_keys'> = type(odict_keys(['new_prefix_DummyMetricSum', 'new_prefix_DummyMetricDiff']))
 +      where odict_keys(['new_prefix_DummyMetricSum', 'new_prefix_DummyMetricDiff']) = keys(keep_base=False)
 +        where keys = MetricCollection(\n  (DummyMetricSum): DummyMetricSum()\n  (DummyMetricDiff): DummyMetricDiff(),\n  prefix=new_prefix_\n).keys
FAILED unittests/bases/test_collections.py::test_metric_collection_prefix_postfix_args[None-_postfix] - AssertionError: assert False
 +  where False = isinstance(dict_keys(['DummyMetricSum', 'DummyMetricDiff']), <class 'odict_keys'>)
 +    where dict_keys(['DummyMetricSum', 'DummyMetricDiff']) = keys(keep_base=True)
 +      where keys = MetricCollection(\n  (DummyMetricSum): DummyMetricSum()\n  (DummyMetricDiff): DummyMetricDiff(),\n  prefix=new_prefix_,\n  postfix=_postfix\n).keys
 +    and   <class 'odict_keys'> = type(odict_keys(['new_prefix_DummyMetricSum_postfix', 'new_prefix_DummyMetricDiff_postfix']))
 +      where odict_keys(['new_prefix_DummyMetricSum_postfix', 'new_prefix_DummyMetricDiff_postfix']) = keys(keep_base=False)
 +        where keys = MetricCollection(\n  (DummyMetricSum): DummyMetricSum()\n  (DummyMetricDiff): DummyMetricDiff(),\n  prefix=new_prefix_,\n  postfix=_postfix\n).keys
FAILED unittests/bases/test_collections.py::test_metric_collection_prefix_postfix_args[prefix_-_postfix] - AssertionError: assert False
 +  where False = isinstance(dict_keys(['DummyMetricSum', 'DummyMetricDiff']), <class 'odict_keys'>)
 +    where dict_keys(['DummyMetricSum', 'DummyMetricDiff']) = keys(keep_base=True)
 +      where keys = MetricCollection(\n  (DummyMetricSum): DummyMetricSum()\n  (DummyMetricDiff): DummyMetricDiff(),\n  prefix=new_prefix_,\n  postfix=_postfix\n).keys
 +    and   <class 'odict_keys'> = type(odict_keys(['new_prefix_DummyMetricSum_postfix', 'new_prefix_DummyMetricDiff_postfix']))
 +      where odict_keys(['new_prefix_DummyMetricSum_postfix', 'new_prefix_DummyMetricDiff_postfix']) = keys(keep_base=False)
 +        where keys = MetricCollection(\n  (DummyMetricSum): DummyMetricSum()\n  (DummyMetricDiff): DummyMetricDiff(),\n  prefix=new_prefix_,\n  postfix=_postfix\n).keys
FAILED unittests/wrappers/test_multitask.py::test_error_on_wrong_keys - AssertionError: Regex pattern did not match.
 Regex: "Expected\\ arguments\\ `task_preds`\\ and\\ `task_targets`\\ to\\ have\\ the\\ same\\ keys\\ as\\ the\\ wrapped\\ `task_metrics`\\.\\ Found\\ task_preds\\.keys\\(\\)\\ =\\ dict_keys\\(\\['Classification'\\]\\),\\ task_targets\\.keys\\(\\)\\ =\\ dict_keys\\(\\['Classification',\\ 'Regression'\\]\\)\\ and\\ self\\.task_metrics\\.keys\\(\\)\\ =\\ odict_keys\\(\\['Classification',\\ 'Regression'\\]\\)"
 Input: "Expected arguments `task_preds` and `task_targets` to have the same keys as the wrapped `task_metrics`. Found task_preds.keys() = dict_keys(['Classification']), task_targets.keys() = dict_keys(['Classification', 'Regression']) and self.task_metrics.keys() = dict_keys(['Classification', 'Regression'])"
```

**Comment by SkafteNicki** [comment]
Failing tests most likely related to #2789

File: .github/workflows/ci-tests.yml [hunk 2]
```diff
@@ -32,7 +32,7 @@ jobs:
       fail-fast: false
       matrix:
         os: ["ubuntu-20.04"]
-        python-version: ["3.9"]
+        python-version: ["3.10"]
```
Comment (Borda) [inline .github/workflows/ci-tests.yml:35]:
```suggestion
        python-version: ["3.9"]
```
seems that we gave some failing tests early PT and newer python

**Comment by Borda** [comment]
> Failing tests most likely related to #2789

I see how about merging this with failing tests for 2.5 but enabling the testing so @bfolie can send a PR which will be validated straight away :)

**Comment by Borda** [comment]
> I prefer the other PR is open before we merge this so we can instantly evaluate if anything else is missing. Else let me check locally if anything else is missing

True but pt 2.5 is already out so any user will hit it... =( my thinking is merging this would prioritise PRs fixing master... Also that accidently won't merge anything that would make it worse... =)

**Pull request merged**